trusted_shuffler/server/src/http.rs

```diff
@@ -75,127 +78,130 @@ impl RequestHandler for HttpRequestHandler {
                 "Couldn't receive response from the backend: {:?}",
                 error
             )),
-            Ok(response) => Ok(hyper_to_trusted_shuffler_response(response).await?),
+            Ok(response) => Ok(HyperResponseWrapper(response).strip().await?),
         }
     }
 }
 
-// We keep the body and the URI from the `hyper::Request`. We convert the body to `Vec`, so it has
-// to be read fully.
-async fn hyper_to_trusted_shuffler_request(
-    hyper_request: Request<Body>,
-) -> anyhow::Result<EncryptedRequest> {
-    let (parts, body) = hyper_request.into_parts();
-    let body = hyper::body::to_bytes(body)
-        .await
-        .context("Couldn't read request body")?;
-    log::debug!("Body {:?}", body);
-
-    let uri = parts.uri.clone();
-
-    // Keep the minimal headers the backend requires to handle the request.
-    let minimal_keys = vec![
-        header::CONTENT_TYPE, // for gRPC requests
-        header::HeaderName::from_static(
-            "grpc-accept-encoding", // for gRPC requests
-        ),
-        header::TE, // for gRPC requests
-        header::CONTENT_LENGTH,
-    ];
-    let minimal_headers = copy_selected_keys(parts.headers, minimal_keys);
-
-    let trusted_shuffler_request = EncryptedRequest {
-        body: body.to_vec(),
-        headers: minimal_headers,
-        uri,
-    };
-
-    Ok(trusted_shuffler_request)
-}
+struct HyperRequestWrapper(Request<Body>);
+
+impl HyperRequestWrapper {
+    // Apart from setting the body and the URI, for our test backend setting to HTTP/2 and POST
```

Comment:
nit: this comment should probably be inside the function as it does not seem to be a doc comment.